mantidproject/mantid
# Added compare workspace tolerance

**Description of work.**
implemented comparing `tableworkspaces` with tolerance, by changing to comparing columns instead of rows, and adding an equality (with tolerance) function to Column. added tolerances to the failing enginX tests.


**To test:**
Run the EnginXScript system tests on windows, they should all pass. Ensure all the unit tests for TableColumn and VectorColumn pass.

Fixes #24546 
Also fixes failing system tests on windows and mac

*This does not require release notes* because **`compareworkspaces` is an algorithm used for testing**


---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
